src/components/ActionModal/ActionOptionsModal.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { SafeAreaView } from 'react-navigation';
+import styled from 'styled-components/native';
+import SlideModal from 'components/Modals/SlideModal';
+import { BaseText } from 'components/Typography';
+import ListItemWithImage from 'components/ListItem/ListItemWithImage';
+import { themedColors } from 'utils/themes';
+import { spacing } from 'utils/variables';
+
+type ItemType = {
+  label: string,
+  key: string,
+  iconName?: string,
+  onPress: () => void,
+  value?: string,
+  isDisabled?: boolean,
+  hide?: boolean,
+}
+
+type Props = {
+  onModalClose: (Function) => void,
+  isVisible: boolean,
+  items: ItemType[],
+  doNotCloseOnPress?: boolean,
+  title?: string,
+};
+
+const MainContainer = styled.View`
+  padding: 22px ${spacing.layoutSides}px 50px;
+`;
+
+class ActionOptionsModal extends React.Component<Props> {
+  renderItem = (item: ItemType) => {
+    const { onModalClose, doNotCloseOnPress } = this.props;
+    const {
+      label,
+      onPress,
+      isDisabled,
+      iconName,
+      key,
+    } = item;
+    return (
+      <ListItemWithImage
+        key={key}
+        customLabel={(
+          <BaseText medium>{label}</BaseText>
+        )}
+        disabled={isDisabled}
+        onPress={() => doNotCloseOnPress ? onPress() : onModalClose(onPress)}
+        diameter={48}
+        iconName={iconName}
+        iconColor={themedColors.link}
+        iconDiameter={24}
+        padding={`5px ${spacing.layoutSides}px`}
+      />
+    );
+  };
+
+  render() {
+    const {
+      onModalClose, isVisible, items, title,
+    } = this.props;
+
+    return (
+      <SlideModal
+        isVisible={isVisible}
+        noClose
+        hideHeader
+        onModalHide={onModalClose}
+      >
+        <SafeAreaView>
+          <BaseText style={{ paddingTop: 24, fontWeight: '500' }} center big>{title}</BaseText>
```

Comment:
Since `title` is optional (based on types), what do you think about conditional rendering here? :)

Also, `fontWeight` as a style isn't working here. We are using `MediumText` component for 500 weight texts.